### PR TITLE
Final Capstone project: We Fly Digital

### PR DESCRIPTION
**Pull request by ibr5500**
In this Pull Request we (@GOlivierNation, @NatiG25, @KanoCode, and @ibr5500 ) have:

- The user logs in to the website, only by typing the username (a properly authenticated login is required if your group is made of 4 people).
- In the navigation panel, the user can see links to:
    - Motorcycles/doctors/classes/items you selected as a theme.
    - "Reserve" form.
    - "My reservations".
    - "Add motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
    - "Delete motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
- On the main page, the user can see a [list of motorcycles/doctors/classes/items that you selected as a theme](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005577).
- When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description (skip the "Rotate image" button).
    - On the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).
- When the user clicks the "Add Item" link in the navigation panel they can see a form for adding a new item.

- Make the app responsive, creating both mobile and desktop versions.
- When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with a title and a "Delete" button.
    - When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- To reserve an appointment, the user has to select a date and city (username and selected item are auto-filled).
    - Use the design based on the ["Book a Vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
    - The user can also access the "Reserve" page from the navigation panel. In that case, only the username is auto-filled.
- When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date, and city).
- Add complete documentation for your API.

[Front-end pull request](https://github.com/ibr5500/flight_reservation-frontend/pull/10)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,148 @@
...
+- Run the app
+```sh
+# Load the schema
+rails db:schema:load
+# Create the database
+rake db:create
+# Run database Migration
+rails db:migrate
+```
+### Usage
```
Comment (mcrd25) [inline README.md:85]: Some of these instrucitons are not necessarily tellign me how to run this backed app.
From cloning to running app with rails server,  I should be able to follow instructions and run application.

Where does `npm i` come in with this app?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    DatabaseCleaner.clean
+  end
+  config.before(:all) do
+    DatabaseCleaner.start
+  end
+  config.after(:all) do
+    DatabaseCleaner.clean
+  end
+  config.infer_spec_type_from_file_location!
+end
```
Comment (mcrd25) [inline spec/rails_helper.rb:32]: I get the following error when running rspec:
image

See [this link](https://stackoverflow.com/questions/61084374/databasecleaner-fails-on-transaction) for reference. Does this issue only occur for me or does it also happen on anyone else's end? Either way, this should help solve

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team (@ibr5500 @GOlivierNation @NatiG25 @KanoCode ), 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there! Really, yo uare almost there!



## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Adding model/unit tests


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Meltrust — APPROVED** [review]
Hi @ibr5500,


## Wow, you did it 🎉
Brilliant
## Thank you for the implemented  changes 💪 🥇 ㊗️ 
#### Unless you want to add more features, go ahead to your final presentation ▶️ ▶️ ▶️ 
### You are about to finish the Microverse program, you have come a long way!!!  Congratulations, and good luck in the software industry.  I'll see you there! 🎉 


### To highlight

- Tests are passing ✔️ 
- Readme was corrected ✔️ 


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

**Comment by TEMESGEN-G25** [comment]
@Meltrust Thank you so much for your uplifting review :smile: & @mcrd25 Thank you for all your suggestions :+1:.



**Note**: All optional & required changes have been implemented.

**Pull request merged**